### ISSUE TITLE
Failed installation : Illegal instruction (core dumped) on Linux ARM64

### ISSUE THREAD

**Issue by dennis-glx** [aws/aws-cli#8008]
### Describe the bug

After following the [instructions](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) for installation on Linux ARM, I get the following error on trying to invoke the CLI tool. 

```bash
$ aws --version
Illegal instruction (core dumped)
```

### Expected Behavior
Expected to get the installed CLI version.

### Current Behavior

```bash
$ aws --version
Illegal instruction (core dumped)
```

Some information of the system: 
```
$ uname -r
Linux agx 4.9.253-tegra #1 SMP PREEMPT Mon Jul 26 12:19:28 PDT 2021 aarch64 aarch64 aarch64 GNU/Linux
```

### GDB Output

```gdb
$  gdb /usr/local/aws-cli/v2/current/dist/aws

GNU gdb (Ubuntu 8.1.1-0ubuntu1) 8.1.1

Reading symbols from /usr/local/aws-cli/v2/current/dist/aws...(no debugging symbols found)...done.
(gdb) run
Starting program: /usr/local/aws-cli/v2/dist/aws
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/aarch64-linux-gnu/libthread_db.so.1".
process 11080 is executing new program: /usr/local/aws-cli/v2/dist/aws
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/aarch64-linux-gnu/libthread_db.so.1".

Program received signal SIGILL, Illegal instruction.
OPENSSL_cpuid_setup () at /aws-crt-python/crt/aws-lc/crypto/fipsmodule/cpucap/cpu_aarch64_linux.c:132
132	/aws-crt-python/crt/aws-lc/crypto/fipsmodule/cpucap/cpu_aarch64_linux.c: No such file or directory.
```

### CLI version used

Latest as of 28 June '23

### Environment details (OS name and version, etc.)

Linux agx 4.9.253-tegra #1 SMP PREEMPT Mon Jul 26 12:19:28 PDT 2021 aarch64 aarch64 aarch64 GNU/Linux

**Comment by tim-finnigan** [comment]
Hi @dennis-glx thanks for reaching out. I think the `Illegal instruction (core dumped)` could indicate a compatibility issue with your architecture. Have you tried installing on another system? It might also be worth opening in an issue in the [aws-crt-python](https://github.com/awslabs/aws-crt-python/issues) repository given the output shared above.

**Comment by andrewhop** [comment]
What CPU are you attempting to install the CLI on, can you run `lscpu` and attach the output?

**Comment by andrewhop** [comment]
Based on the line number `cpu_aarch64_linux.c:132` I believe you are hitting the same MIDR_EL1 issue that Valgrind was hitting. This was fixed in https://github.com/aws/aws-lc/pull/1006 which was released in [v1.10.0](https://github.com/aws/aws-lc/releases/tag/v1.10.0) of AWS-LC. The most recent version of the [Python CRT](https://github.com/awslabs/aws-crt-python/tree/fc7074dc36eb634ccedac5cb0c92a93af71c7d41/crt) includes [aws-lc @ 80c3943](https://github.com/awslabs/aws-lc/tree/80c394324382c78d2cbd783babaad1f612c1642d) which is [v1.11.0](https://github.com/aws/aws-lc/releases/tag/v1.11.0) which includes the fix. Can you try updating to a newer version of the AWS CLI?

**Comment by Dennis-Johnson** [comment]
Sure. 

```bash
$ lscpu 

Architecture:                    aarch64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
CPU(s):                          8
On-line CPU(s) list:             0-3
Off-line CPU(s) list:            4-7
Thread(s) per core:              1
Core(s) per socket:              2
Socket(s):                       2
Vendor ID:                       Nvidia
Model:                           0
Model name:                      ARMv8 Processor rev 0 (v8l)
Stepping:                        0x0
CPU max MHz:                     2265.6001
CPU min MHz:                     115.2000
BogoMIPS:                        62.50
L1d cache:                       256 KiB
L1i cache:                       512 KiB
L2 cache:                        4 MiB
L3 cache:                        4 MiB
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Not affected
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Branch predictor hardening
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm aes pmull sha1 sha2 crc32 atom
                                 ics fphp asimdhp cpuid asimdrdm dcpop

```

**Comment by Dennis-Johnson** [comment]
If it helps, I ran the test code from the LC repo issue you mentioned. 
```cpp
#include <iostream>
#include <cstdint>
#include <asm/hwcap.h>
#include <sys/auxv.h>

static uint64_t armv8_cpuid_probe() {
  uint64_t val;
  __asm__ volatile("mrs %0, MIDR_EL1" : "=r" (val));
  return val;
}

int main() {
  uint32_t hwcap = getauxval(AT_HWCAP);
  std::cout << "hwcap = " << hwcap << "\n";
  std::cout << "HWCAP_CPUID = " << HWCAP_CPUID << "\n";
  uint32_t has_cpu_id = hwcap & HWCAP_CPUID;
  std::cout << "hwcap & HWCAP_CPUID = " << has_cpu_id << "\n";
  if (has_cpu_id) {
    std::cout << "armv8_cpuid_probe = " << armv8_cpuid_probe() << "\n";
  }
  return 0;
}
```

### Output
```bash
hwcap = 73727
HWCAP_CPUID = 2048
hwcap & HWCAP_CPUID = 2048
armv8_cpuid_probe = 1309605952
```

**Comment by ArivCR7** [comment]
@andrewhop , I'm also getting the same issue. This is impacting our production. Can you please investigate and let us know the resolution.

**Comment by tim-finnigan** [comment]
After further research by the team it appears that increasing the awscrt version ceiling should address this. Here is the PR forthat: https://github.com/aws/aws-cli/pull/8056. A related PR (https://github.com/aws/aws-lc/pull/1006) was referenced in an [earlier comment](https://github.com/aws/aws-cli/issues/8008#issuecomment-1624296636) as well as awscrt [v0.16.18](https://github.com/awslabs/aws-crt-python/releases/tag/v0.16.18).

**Comment by amberkushwaha** [comment]
after further research by the team it appears that the corrupted coding is now being settled by the clients end and also the mono-log was shifted to the server.it seems the issue is been settled now and if it reappears i will handle it by sanctioning it to the core team and log.

**Comment by amberkushwaha** [comment]
update awscrt requirements by the team it appears that the corrupt coding is now being settled by the clients end and also thr mono log was shifted to the servent it seems the issue is been settled now and if it reappears i will handle it by sanctioning it to the core team in it and remaining log modules.

**Comment by tim-finnigan** [comment]
The `awscrt` ceiling was raised to 0.19.12 in https://github.com/aws/aws-cli/pull/8324. This should now be resolved in v2.13.35+ ([CHANGELOG](https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst?plain=1) for reference.) If anyone is still seeing this or related issues please let us know. Thanks!
